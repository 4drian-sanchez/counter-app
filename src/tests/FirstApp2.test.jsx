/** @jest-environment jsdom */

import { getByRole, render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Pruebas en <FirstApp/>', () => { 
   const title = 'Hola soy goku'
   const  subTitle  = 'soy un subttitulo'

   test('Debe de hacer match en el snapshot', () => { 
      const { container  } = render( <FirstApp title={title} subTitle={subTitle}/> )
      expect( container ).toMatchSnapshot();
    })

   test('Debe de mostrar en mensaje, hola soy Goku', () => { 
      const { container  } = render( <FirstApp title={title} subTitle={subTitle}/> )
      expect( screen.getByText(title)).toBeTruthy();
    })

   test('Debe de mostrar el titulo en un h1', () => { 
      const { container  } = render( <FirstApp title={title} subTitle={subTitle}/> )
      expect( screen.getByRole( "heading", {level: 1}).textContent ).toContain(title);
    })
 })