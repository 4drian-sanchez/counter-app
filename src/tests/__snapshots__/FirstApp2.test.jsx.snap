// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe de hacer match en el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy goku
  </h1>
  <p>
    Hola adrian soy tu del pasado
  </p>
  <p>
    hola mundo, todo bonito o que?
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Laudantium pariatur, voluptates dolor adipisci asperiores illum iure consequatur beatae neque tempora.
  </p>
  <h2>
    soy un subttitulo
  </h2>
  <p>
     
    adrian
  </p>
</div>
`;
